### PR TITLE
#2559. Add factory constructors to augmented_expression_A04_t*.dart tests

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Augmentation-libraries/augmented_expression_A04_t07_lib.dart [hunk 1]
```diff
@@ -90,6 +97,13 @@ augment extension Ext {
 }
 
 augment extension type ET {
+  augment factory ET.f() {
+    var [augmented] = [42];
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmented_expression_A04_t07_lib.dart:101]: Should be a map pattern?

File: LanguageFeatures/Augmentation-libraries/augmented_expression_A04_t25_lib.dart [hunk 2]
```diff
@@ -26,6 +26,12 @@ augment String topLevelFunction() {
 }
 
 augment class C {
+  augment factory C.f() {
+    var ag = augmented();
+    Expect.equals("Original",ag.log);
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmented_expression_A04_t25_lib.dart:31]: Funny spacing, format?

Comment (eernstg) [reply LanguageFeatures/Augmentation-libraries/augmented_expression_A04_t25_lib.dart:31]: Some other cases similar.

File: LanguageFeatures/Augmentation-libraries/augmented_expression_A04_t28_lib.dart [hunk 3]
```diff
@@ -39,6 +39,22 @@ augment String topLevelFunction4({required String value}) =>
 // [cfe] unspecified
 
 augment class C {
+  augment factory C.f1(String value) => augmented("a", "b");
+//                                     ^^^^^^^^^
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmented_expression_A04_t28_lib.dart:43]: Align carets?

**Review by eernstg — COMMENTED** [review]
Looks good, couple of typos.

File: LanguageFeatures/Augmentation-libraries/augmented_expression_A04_t07_lib.dart (hunk 1 above)
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmented_expression_A04_t07_lib.dart]: Of course! Thank you!

File: LanguageFeatures/Augmentation-libraries/augmented_expression_A04_t25_lib.dart (hunk 2 above)
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmented_expression_A04_t25_lib.dart]: Fixed

File: LanguageFeatures/Augmentation-libraries/augmented_expression_A04_t28_lib.dart (hunk 3 above)
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmented_expression_A04_t28_lib.dart]: Aligned

**Review by sgrekhov — COMMENTED** [review]
Thank you! Fixed. PTAL

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**